File: app/src/main/java/ua/com/motometer/android/core/facade/api/model/ExpenseRecord.java
package ua.com.motometer.android.core.facade.api.model;

import org.immutables.value.Value;

import java.math.BigDecimal;
import java.time.LocalDate;

@Value.Immutable
public interface ExpenseRecord {
    @Value.Parameter
    long id();

    @Value.Parameter
    BigDecimal amount();

    @Value.Parameter
    LocalDate date();

    @Value.Parameter
    String description();
}

File: app/src/main/java/ua/com/motometer/android/core/facade/api/model/ExpenseRecord.kt
package ua.com.motometer.android.core.facade.api.model

import org.immutables.value.Value

import java.math.BigDecimal
import java.time.LocalDate

@Value.Immutable
interface ExpenseRecord {

    @Value.Parameter
    fun amount(): BigDecimal

    @Value.Parameter
    fun date(): LocalDate

    @Value.Parameter
    fun description(): String
}
